### ISSUE TITLE
Add fan-out scanning factory support

### ISSUE THREAD

**Issue by JonZeolla** [SeisoLLC/easy_infra#301]
# Summary of the feature

Currently there is no built-in capability to run a fan-out, parallel scan all of the terraform in a supported VCS. For instance, a company may have 5,000 repositories in GitLab, and 1,200 of them contain terraform. We should support a way to clone a provided list of repositories in parallel (potentially in multiple runtimes environments, like separate lambdas or EC2 instances), scan the repositories for matching file extensions, run the appropriate security tools against the correct folders in that repository, and centralize the results.

Out of scope is the discovery/enumeration logic of which repositories exist; that should be handled by other projects.
